From 892f8adade10641f1f04f04b589753d928fb5443 Mon Sep 17 00:00:00 2001
From: Doug Torrance <dtorrance@piedmont.edu>
Date: Sun, 9 May 2021 12:14:19 -0400
Subject: [PATCH] Use std::ptrdiff_t for vector iterator difference type

It was previously long (64-bit integers), but this will not be the
case on 32-bit systems.

Closes: #273
---
 linbox/vector/bit-vector.inl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/linbox/vector/bit-vector.inl b/linbox/vector/bit-vector.inl
index 5fa84b421..23a22ba47 100644
--- a/linbox/vector/bit-vector.inl
+++ b/linbox/vector/bit-vector.inl
@@ -46,7 +46,7 @@ namespace std
 		typedef LinBox::BitVector::reference reference;
 		typedef bool *pointer;
 		typedef bool value_type;
-		typedef long difference_type;
+		typedef std::ptrdiff_t difference_type;
 	};
 
 	template <>
@@ -56,7 +56,7 @@ namespace std
 		typedef LinBox::BitVector::const_reference reference;
 		typedef const bool *pointer;
 		typedef bool value_type;
-		typedef long difference_type;
+		typedef std::ptrdiff_t difference_type;
 	};
 }
 
